Mathlib/Probability/Distributions/Pareto.lean

```diff
@@ -0,0 +1,151 @@
+
+lemma paretoPDF_eq (t r x : ℝ) :
+    paretoPDF t r x = ENNReal.ofReal (if t ≤ x then r * t ^ r * x ^ (-(r + 1)) else 0) := rfl
+
+lemma paretoPDF_of_lt {t r x : ℝ} (hx : x < t) : paretoPDF t r x = 0 := by
+  simp only [paretoPDF_eq, if_neg (not_le.mpr hx), ENNReal.ofReal_zero]
+
+lemma paretoPDF_of_le {t r x : ℝ} (hx : t ≤ x) :
+    paretoPDF t r x = ENNReal.ofReal (r * t ^ r * x ^ (-(r + 1))) := by
+  simp only [paretoPDF_eq, if_pos hx]
+
+/-- The Lebesgue integral of the Pareto pdf over reals ≤ t equals 0. -/
+lemma lintegral_paretoPDF_of_le {t r x : ℝ} (hx : x ≤ t) :
+    ∫⁻ y in Iio x, paretoPDF t r y = 0 := by
+  rw [setLIntegral_congr_fun (g := fun _ ↦ 0) measurableSet_Iio]
+  · rw [lintegral_zero, ← ENNReal.ofReal_zero]
+  · simp only [paretoPDF_eq, ge_iff_le, ENNReal.ofReal_eq_zero]
+    filter_upwards with a (_ : a < _)
+    rw [if_neg (by linarith)]
+
+/-- The Pareto pdf is measurable. -/
+@[measurability, fun_prop]
+lemma measurable_paretoPDFReal (t r : ℝ) : Measurable (paretoPDFReal t r) :=
+  Measurable.ite measurableSet_Ici ((measurable_id.pow_const _).const_mul _) measurable_const
+
+/-- The Pareto pdf is strongly measurable. -/
+@[measurability]
+ lemma stronglyMeasurable_paretoPDFReal (t r : ℝ) :
+     StronglyMeasurable (paretoPDFReal t r) :=
+   (measurable_paretoPDFReal t r).stronglyMeasurable
+
+/-- The Pareto pdf is positive for all reals >= t. -/
+lemma paretoPDFReal_pos {t r x : ℝ} (ht : 0 < t) (hr : 0 < r) (hx : t ≤ x) :
+    0 < paretoPDFReal t r x := by
+  rw [paretoPDFReal, if_pos hx]
+  have _ : 0 < x := by linarith
+  positivity
+
+/-- The Pareto pdf is nonnegative. -/
+lemma paretoPDFReal_nonneg {t r : ℝ} (ht : 0 ≤ t) (hr : 0 ≤ r) (x : ℝ) :
+    0 ≤ paretoPDFReal t r x := by
+  unfold paretoPDFReal
+  split_ifs with h
+  · cases le_iff_eq_or_lt.1 ht with
+    | inl ht0 =>
+      rw [← ht0] at h
+      positivity
+    | inr htp =>
+      have := lt_of_lt_of_le htp h
+      positivity
+  · positivity
+
+open Measure
+
+/-- The pdf of the Pareto distribution integrates to 1. -/
+@[simp]
+lemma lintegral_paretoPDF_eq_one {t r : ℝ} (ht : 0 < t) (hr : 0 < r) :
+    ∫⁻ x, paretoPDF t r x = 1 := by
+  have leftSide : ∫⁻ x in Iio t, paretoPDF t r x = 0 := lintegral_paretoPDF_of_le (le_refl t)
+  have rightSide : ∫⁻ x in Ici t, paretoPDF t r x =
+      ∫⁻ x in Ici t, ENNReal.ofReal (r * t ^ r * x ^ (-(r + 1))) :=
+    setLIntegral_congr_fun measurableSet_Ici (ae_of_all _ (fun _ ↦ paretoPDF_of_le))
+  rw [← ENNReal.toReal_eq_one_iff, ← lintegral_add_compl _ measurableSet_Ici, compl_Ici,
+    leftSide, rightSide, add_zero, ← integral_eq_lintegral_of_nonneg_ae]
+  · rw [integral_Ici_eq_integral_Ioi, integral_mul_left, integral_Ioi_rpow_of_lt _ ht]
+    · field_simp [hr]
+      rw [mul_assoc, ← rpow_add ht]
+      simp
+    linarith
+  · rw [EventuallyLE, ae_restrict_iff' measurableSet_Ici]
+    exact ae_of_all _ (fun x (hx : t ≤ x) ↦ by
+      have _ : 0 < x := by linarith
+      positivity
+    )
```

Comment:
Please avoid multiline tactic blocks in the arguments of a theorem. Use `apply` or `refine` instead.
```suggestion
   refine ae_of_all _ fun x (hx : t ≤ x) ↦ ?_
    have := lt_of_lt_of_le ht hx
    positivity
```